crawler/helpers/logging_helpers.py

```diff
@@ -322,6 +322,77 @@ def __init__(self):
         self.short_display_description = "Unknown date format"
 
 
+class AggregateType28(AggregateTypeBase):
+    def __init__(self):
+        super().__init__()
+        self.type_str = "TYPE 28"
+        self.error_level = ErrorLevel.CRITICAL
+        self.message = (
+            f"{self.error_level.name}: Samples from priority where the MLWH database "
+            f"insert has failed. ({self.type_str})"
+        )
+        self.short_display_description = "Failed MLWH inserts"
+
+
+class AggregateType29(AggregateTypeBase):
+    def __init__(self):
+        super().__init__()
+        self.type_str = "TYPE 29"
+        self.error_level = ErrorLevel.CRITICAL
+        self.message = (
+            f"{self.error_level.name}: Samples from priority where the MLWH database connection "
+            f"could not be made. ({self.type_str})"
+        )
+        self.short_display_description = "Failed MLWH connection"
+
+
+class AggregateType30(AggregateTypeBase):
+    def __init__(self):
+        super().__init__()
+        self.type_str = "TYPE 30"
+        self.error_level = ErrorLevel.CRITICAL
+        self.message = (
+            f"{self.error_level.name}: Priority samples where all DART database inserts have failed. ({self.type_str})"
+        )
+        self.short_display_description = "Failed DART priority samples inserts"
+
+
+class AggregateType31(AggregateTypeBase):
+    def __init__(self):
+        super().__init__()
+        self.type_str = "TYPE 31"
+        self.error_level = ErrorLevel.CRITICAL
+        self.message = (
+            f"{self.error_level.name}: Priority samples where the DART database connection "
+            f"could not be made. ({self.type_str})"
+        )
+        self.short_display_description = "Failed priority samples DART connection"
+
+
+class AggregateType32(AggregateTypeBase):
+    def __init__(self):
+        super().__init__()
+        self.type_str = "TYPE 32"
+        self.error_level = ErrorLevel.CRITICAL
+        self.message = (
+            f"{self.error_level.name}: Priority samples that we have in Mongodb "
+            f"but they are still unprocessed. ({self.type_str})"
+        )
+        self.short_display_description = "Step 2 validation failure"
```

Comment:
Add a final validation to include this error

Reply:
check also that after step 1 there are no unprocessed priority samples for the given sample ids?
check after step 2 the above, that there are no unprocessed priority samples